src/test/java/org/sqlite/SerializeTest.java

```diff
@@ -0,0 +1,245 @@
+            execute(connection, "CREATE TABLE a_schema.a_table (x integer)");
+            execute(connection, "INSERT INTO a_schema.a_table (x) values (?)", 1007);
+            int pageSize = fetch(connection, "pragma a_schema.page_size");
+            int pageCount = fetch(connection, "pragma a_schema.page_count");
+            assertThat(connection.serializeSize("a_schema")).isEqualTo((long) pageSize * pageCount);
+        }
+    }
+
+    @Test
+    public void testBufferIsFileCompatible() throws SQLException, IOException {
+
+        byte[] buff = serialize();
+        File tmp = File.createTempFile("sqlite_test", ".db");
+        Files.write(tmp.toPath(), buff);
+        try (SQLiteConnection connection =
+                (SQLiteConnection)
+                        DriverManager.getConnection("jdbc:sqlite:" + tmp.getAbsolutePath())) {
+            assertThat(fetch(connection, "SELECT * FROM main.a_table")).isEqualTo(1007);
+        }
+        assertThat(tmp.delete()).isTrue();
+    }
+
+    @Test
+    public void testFileIsBufferCompatible() throws SQLException, IOException {
+        File tmp = File.createTempFile("sqlite_test", ".db");
+        try (SQLiteConnection connection =
+                (SQLiteConnection)
+                        DriverManager.getConnection("jdbc:sqlite:" + tmp.getAbsolutePath())) {
+            execute(connection, "CREATE TABLE a_table (x integer)");
+            execute(connection, "INSERT INTO a_table (x) values (?)", 1007);
+        }
+        byte[] buff = Files.readAllBytes(tmp.toPath());
+        deserializeAndAssert(buff);
+        assertThat(tmp.delete()).isTrue();
+    }
+
+    @Disabled("This takes around 15 seconds on a fast machine and consumes 4gb of memory")
+    @Test
+    public void testVeryLarge() throws SQLException {
+        byte[] large;
+        int rowCount;
+
+        try (SQLiteConnection connection =
+                (SQLiteConnection) DriverManager.getConnection("jdbc:sqlite:")) {
+            execute(connection, "ATTACH ? AS ?", ":memory:", "a_schema");
+            connection.deserialize("a_schema", serialize());
+
+            StringBuilder sb = new StringBuilder();
+            for (int i = 0; i < 500; ++i) { // we want this to be well withing a page
+                sb.append("a");
+            }
+            String s = sb.toString();
+            assertThatThrownBy(
+                            () -> {
+                                //noinspection InfiniteLoopStatement
+                                while (true) {
+                                    execute(
+                                            connection,
+                                            "INSERT INTO a_schema.a_table (x) values (?)",
+                                            s);
+                                }
+                            })
+                    .isInstanceOf(SQLException.class)
+                    .hasFieldOrPropertyWithValue("errorCode", SQLiteErrorCode.SQLITE_FULL.code);
```

Comment:
```suggestion
                    .asInstanceOf(InstanceOfAssertFactories.type(SQLException.class))
                    .extracting(SQLException::getErrorCode)
                    .isEqualTo(SQLiteErrorCode.SQLITE_FULL.code);
```

Reply:
did it a bit different

Reply:
That's fine, one minor note: All the `isInstanceOf` checks should probably check against `SQLiteException` and not `SQLException` to be in line with the casts to `SQLiteException`.

Reply:
I started of testing against the contract, but you are right, if I cast, I should check it correctly. done.